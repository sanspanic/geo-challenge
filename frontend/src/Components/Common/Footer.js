import React, { useContext } from "react";
import {
  TwitterLogo,
  GithubLogo,
  PencilLine,
  LinkedinLogo,
} from "phosphor-react";
import GameContext from "../../Context/GameContext";

const Footer = () => {
  const { status } = useContext(GameContext);
  let isHidden = null;
  if (status.isActive) isHidden = "hidden";
  return (
    <>
      <footer
        className={`${isHidden} footer mt-6 text-gray-700 border-t body-font`}
      >
        <div className="bg-black">
          <div className="container flex flex-col flex-wrap px-5 py-6 mx-auto lg:px-20 sm:flex-row">
            <span className="inline-flex justify-center mt-2 sm:mt-0 sm:justify-start">
              <a
                href="https://twitter.com/sansPanicDev"
                className="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
              >
                <TwitterLogo size={24} />
              </a>
              <a
                href="https://github.com/sanspanic"
                className="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
              >
                <GithubLogo size={24} />
              </a>
              <a
                href="https://dev.to/sanspanic"
                className="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
              >
                <PencilLine size={24} />
              </a>
              <a
                href="https://www.linkedin.com/in/sandra-spanikova-6738aa1b7/"
                className="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
              >
                <LinkedinLogo size={24} />
              </a>
            </span>
            <span className="font-mono text-center py-2 sm:ml-auto text-xs text-gradient-purple">
              Made by sansPanic. I write code and I write about code.
            </span>
          </div>
        </div>
      </footer>
    </>
  );
};

export default Footer;
