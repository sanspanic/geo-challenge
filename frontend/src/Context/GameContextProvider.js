import React, { useState, useEffect } from "react";
import GameContext from "./GameContext";
import geoApi from "../API/geoAPI";

const GameContextProvider = ({ children }) => {
  const [speedBonus, setSpeedBonus] = useState(0);
  const [width, setWidth] = useState(100);
  const [level, setLevel] = useState(1);
  const [score, setScore] = useState(0);
  const [countries, setCountries] = useState([]);
  const [rank, setRank] = useState(1);
  const [hasLoaded, setHasLoaded] = useState(false);
  const [lives, setLives] = useState(5);
  const [finalScore, setFinalScore] = useState(0);

  const [status, setStatus] = useState({
    isActive: false,
    isLost: true,
    isWon: false,
  });

  useEffect(() => {
    const getCountryData = async () => {
      try {
        const res = await geoApi.getData();
        setCountries(res);
        setHasLoaded(true);
      } catch (err) {
        console.log(err);
      }
    };
    getCountryData();
  }, []);

  return (
    <GameContext.Provider
      value={{
        level,
        setLevel,
        score,
        setScore,
        countries,
        setCountries,
        hasLoaded,
        status,
        setStatus,
        lives,
        setLives,
        width,
        setWidth,
        speedBonus,
        setSpeedBonus,
        rank,
        setRank,
        finalScore,
        setFinalScore,
      }}
    >
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
