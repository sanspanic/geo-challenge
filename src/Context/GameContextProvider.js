import React, { useState } from "react";
import GameContext from "./GameContext";

const GameContextProvider = ({ children }) => {
  const [level, setLevel] = useState({ num: 1, title: "Guess The Flag" });
  const [score, setScore] = useState(0);

  return (
    <GameContext.Provider value={{ level, setLevel, score, setScore }}>
      {children}
    </GameContext.Provider>
  );
};

export default GameContextProvider;
