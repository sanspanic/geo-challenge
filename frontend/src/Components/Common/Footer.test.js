import React from "react";
import { render, act } from "@testing-library/react";
import Footer from "./Footer";
import GameContextProvider from "../../Context/GameContextProvider";

test("renders without crashing", () => {
  render(
    <GameContextProvider>
      <Footer />
    </GameContextProvider>
  );
});

test("matches snapshot", () => {
  const { asFragment } = render(
    <GameContextProvider>
      <Footer />
    </GameContextProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
