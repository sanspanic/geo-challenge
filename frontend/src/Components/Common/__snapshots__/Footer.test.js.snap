// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="null footer mt-6 text-gray-700 border-t body-font"
  >
    <div
      class="bg-black"
    >
      <div
        class="container flex flex-col flex-wrap px-5 py-6 mx-auto lg:px-20 sm:flex-row"
      >
        <span
          class="inline-flex justify-center mt-2 sm:mt-0 sm:justify-start"
        >
          <a
            class="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
            href="https://twitter.com/sansPanicDev"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M48,200s40-8,48-32c0,0-64-24-48-112,0,0,32,40,80,48V88.00288a40.00668,40.00668,0,0,1,76.67148-16.00327L240,72l-32,32c0,56-40,112-112,112C64,216,48,200,48,200Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </a>
          <a
            class="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
            href="https://github.com/sanspanic"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M84,232a24,24,0,0,0,24-24V160"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M172,232a24,24,0,0,1-24-24V160"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M152,160h16a24,24,0,0,1,24,24v8a24,24,0,0,0,24,24"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M104,160H88a24,24,0,0,0-24,24v8a24,24,0,0,1-24,24"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <path
                d="M64.51166,76.70377A51.90056,51.90056,0,0,1,68,32a51.9599,51.9599,0,0,1,43.82469,23.9988V56h32.35061V55.9988A51.9599,51.9599,0,0,1,188,32a51.90056,51.90056,0,0,1,3.48834,44.70377l0,0A47.77872,47.77872,0,0,1,200,104v8a48,48,0,0,1-48,48H104a48,48,0,0,1-48-48v-8a47.77872,47.77872,0,0,1,8.51163-27.29627Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </a>
          <a
            class="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
            href="https://dev.to/sanspanic"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <path
                d="M96,216H48a8,8,0,0,1-8-8V163.31371a8,8,0,0,1,2.34315-5.65686l120-120a8,8,0,0,1,11.3137,0l44.6863,44.6863a8,8,0,0,1,0,11.3137Z"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="136"
                x2="192"
                y1="64"
                y2="120"
              />
              <polyline
                fill="none"
                points="216 216 96 216 40.509 160.509"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
            </svg>
          </a>
          <a
            class="ml-4 text-white hover:text-cerise-500 transition ease-in-out duration-500"
            href="https://www.linkedin.com/in/sandra-spanikova-6738aa1b7/"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 256 256"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                fill="none"
                height="256"
                width="256"
              />
              <rect
                fill="none"
                height="176"
                rx="8"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                width="176"
                x="40"
                y="40"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="120"
                x2="120"
                y1="112.00094"
                y2="176.00094"
              />
              <line
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
                x1="88"
                x2="88"
                y1="112.00094"
                y2="176.00094"
              />
              <path
                d="M120,140.00094a28,28,0,1,1,56,0v36"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="16"
              />
              <circle
                cx="88"
                cy="79.99998"
                r="12"
              />
            </svg>
          </a>
        </span>
        <span
          class="font-mono text-center py-2 sm:ml-auto text-xs text-gradient-purple"
        >
          Made by sansPanic. I write code and I write about code.
        </span>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
